tests/foreman/cli/test_realm.py

```diff
@@ -0,0 +1,327 @@
+# -*- encoding: utf-8 -*-
+# pylint: disable=no-self-use
+"""Test class for Realm CLI
+
+:Requirement: Realm
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: CLI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import random
+
+from fauxfactory import gen_string
+from robottelo.cleanup import capsule_cleanup
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.factory import (
+    make_location,
+    make_proxy,
+    make_realm,
+    CLIFactoryError,
+)
+from robottelo.cli.realm import Realm
+from robottelo.decorators import (
+    tier1,
+)
+from robottelo.test import CLITestCase
+
+
+class RealmTestCase(CLITestCase):
+    """Tests for Realms via Hammer CLI, must be run on QE RHEL7 Sat6.3 host"""
+
+    def _make_proxy(self, options=None):
+        """Create a Proxy and register the cleanup function"""
+        proxy = make_proxy(options=options)
+        # Add capsule to cleanup list
+        self.addCleanup(capsule_cleanup, proxy['id'])
+        return proxy
+
+    @tier1
+    def test_positive_create_delete_name(self):
```

Comment:
Yes, when the function exits if a realm is still attached to the proxy then it will throw an error:

CLIReturnCodeError: CLIReturnCodeError(return_code=65, stderr=u'[ERROR 2017-07-21 08:47:10 API] 422 Unprocessable Entity\n[ERROR 2017-07-21 08:47:10 Exception] qe-testing-rhel7.satqe.lab.eng.rdu2.redhat.com is used by realm\nCould not delete the proxy:\n  qe-testing-rhel7.satqe.lab.eng.rdu2.redhat.com is used by realm\n', msg=u'Command "proxy delete" finished with return_code 65\nstderr contains following message:\n[ERROR 2017-07-21 08:47:10 API] 422 Unprocessable Entity\n[ERROR 2017-07-21 08:47:10 Exception] qe-testing-rhel7.satqe.lab.eng.rdu2.redhat.com is used by realm\nCould not delete the proxy:\n  qe-testing-rhel7.satqe.lab.eng.rdu2.redhat.com is used by realm\n'